opensearch-project/OpenSearch-Dashboards
# [Workspace] Add User level setting

### Description

This PR includes 
1. user settings page
2. ui_setting_client changes to support user personal settings
3. UserUISettingsClientWrapper to translate user id placeholder to real user id


### Issues Resolved




https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7821

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7909

## Screenshot





https://github.com/user-attachments/assets/57e9962b-4869-4a41-911e-9a698480708e



## Testing the changes


1. Run OSD enabled `yarn start --no-base-path --workspace.enabled=true`
2. make sure security is enabled
3. open http://localhost:5601, and then go to dev tools
4. Run kibana search to make sure global setting been created
```
GET .kibana/_search
{
  "query": {
    "bool": {
      "must": [
        {
          "match": {
            "type": "config"
          }
        }
      ]
    }
  }
```
5. Run command to add user setting, after that run kibana query again to verify 
a config doc with id `config:admin` been indexed and it has defaultWorkspace as "ABC"
```
curl --request POST \
  --url http://localhost:5601/api/opensearch-dashboards/settings \
  --header 'Authorization: Basic YWRtaW46YWRtaW4=' \
  --header 'Content-Type: application/json' \
  --header 'osd-xsrf: osd-fetch' \
  --data '{
 "changes": {
  "defaultIndex": "logstash-*",
  "defaultWorkspace": "ABC"
 }
}'
```


## Changelog

- feat: [Workspace] Support user personal settings

### Check List
